src/main/scala/auth/Jwt.scala

```diff
@@ -0,0 +1,36 @@
+package io.blindnet.identityclient
+package auth
+
+import cats.effect.*
+import io.blindnet.jwt.Token
+import io.circe.*
+
+import java.util.UUID
+
+sealed trait Jwt {
+  val appId: UUID
+
+  def asApp: Option[AppJwt] = as(classOf[AppJwt])
```

Comment:
Can be reduced to

```scala
def as[T <: Jwt](using scala.reflect.TypeTest[Jwt, T]): Option[T] =
    this match {
      case t: T => Some(t)
      case _     => None
    }
```
```scala
def requireAppJwt: JwtAuthenticator[AppJwt] = require(_.as[AppJwt])
def requireAnyUserJwt: JwtAuthenticator[AnyUserJwt] = require(_.as[AnyUserJwt])
```

Reply:
Didn't know about this `TypeTest` class, that's cool! I'll change that, but I still think having aliases for the Jwt types is nice in a way (`_.asUser` vs `_.as[JwtUser]`).

Reply:
Agree, we can keep the aliases for different types since there are just a few of them.